modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -168,16 +184,45 @@ private void validateTopLevelParameter(ParameterModel parameter, ParameterizedMo
 
     private Set<String> collectRepeatedNames(List<? extends Named> namedObject) {
       Set<String> names = new HashSet<>();
-      Set<String> repeatedNames =
-          namedObject.stream().filter(parameter -> !names.add(parameter.getName())).map(Named::getName).collect(toSet());
-
-      return repeatedNames;
+      return namedObject.stream()
+          .filter(parameter -> !names.add(parameter.getName()))
+          .map(Named::getName).collect(toSet());
     }
 
     private void validateNameClashes(Collection<? extends Named>... collections) {
       Multimap<String, Named> names = LinkedListMultimap.create();
       stream(collections).flatMap(Collection::stream).forEach(named -> names.put(named.getName(), named));
 
+      Map<String, Named> singularClashes = names
+          .entries()
+          .stream()
+          .filter(e -> singularizedObjects.keySet().contains(e.getKey()))
+          .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+      if (!singularClashes.isEmpty()) {
+        StringBuilder builder = new StringBuilder(format("Extension '%s' contains %d parameters that clash when singularized. ",
+                                                         extensionModel.getName(), singularClashes.size()));
+
+        builder.append(singularClashes.keySet()
+            .stream()
+            .map(k -> {
+              DescribedParameter reference = singularizedObjects.get(k);
+              Collection<Named> namedElements = names.get(k);
+              if (!namedElements.isEmpty()) {
+                Named next = namedElements.iterator().next();
+                return format(
+                              "%s name '%s' contains parameter '%s' that clash when singularized with parameter '%s' in '%s'",
+                              reference.parent.getDescription(), reference.parent.getName(), reference.getName(), k,
+                              next.getName());
+              } else {
```

Comment:
filter this when creating the singularClashes map, and also see if is not better just to have a Map<String, Set> instead of <String, Named>, specially since the value is never used